pkg/security/ebpf/c/include/hooks/mount.h

```diff
@@ -222,12 +222,16 @@ int __attribute__((always_inline)) kprobe_dr_unshare_mntns_stage_one_callback(st
     syscall->resolver.ret = 0;
 
     resolve_dentry(ctx, DR_KPROBE);
+
+    // if the tail call fails, we need to pop the syscall cache entry
+    pop_syscall(EVENT_UNSHARE_MNTNS);
+
     return 0;
 }
 
 SEC("kprobe/dr_unshare_mntns_stage_two_callback")
 int __attribute__((always_inline)) kprobe_dr_unshare_mntns_stage_two_callback(struct pt_regs *ctx) {
-    struct syscall_cache_t *syscall = peek_syscall(EVENT_UNSHARE_MNTNS);
+    struct syscall_cache_t *syscall = pop_syscall(EVENT_UNSHARE_MNTNS);
```

Comment:
I'm not sure the `EVENT_UNSHARE_MNTNS` event should be popped here because this program is called multiple times from the `attach_mnt` kprobe ([kernel code](https://elixir.bootlin.com/linux/v6.3.9/source/fs/namespace.c#L1895)).
A kretprobe may be needed to pop this one.

Reply:
Thanks, I'll use a kret then